### PR TITLE
support for max relayed connections

### LINKED ISSUE #208: Add relay slots / basic QoS
## Context

Hopr nodes that are willing to become a relay for other can announce themself on-chain. Other nodes can then try to contact these nodes in order announce to other nodes that they are available "behind" these nodes.

Since these nodes have limited resources, they should only allow a limited number number of relay requests.

## Desired feature

Define an upper bound of simultaneous relay connection and deny all further relay requests

- Add an additional error, showing the relay "is full"
- Handle this error at the client and try a different relay

### PR DESCRIPTION
**Pull request by andreykiryushkin**
This PR adds `maxRelayedConnections` parameter to `HoprConnect` module effectively turning back the others seeking to relay through it with a graceful error code.

### CODE REVIEW

File: src/relay/handshake.ts [hunk 1]
```diff
@@ -79,6 +82,15 @@ class RelayHandshake {
     this.shaker = handshake<StreamResult>(stream)
   }
 
+  async reject() {
```
Comment (robertkiel) [inline src/relay/handshake.ts:85]: What about `reject(reason)` ?

File: tests/node.ts [hunk 2]
```diff
@@ -169,10 +173,17 @@ async function executeCommands({
...
-          new Multiaddr(`/p2p/${relayPeerId}/p2p-circuit/p2p/${targetPeerId.toB58String()}`),
-          TEST_PROTOCOL
-        )
+        const { stream } = await node
+          .dialProtocol(
+            new Multiaddr(`/p2p/${relayPeerId}/p2p-circuit/p2p/${targetPeerId.toB58String()}`),
+            TEST_PROTOCOL
+          )
+          .catch((err) => {
+            console.log(`dialProtocol to ${cmdDef.targetIdentityName} failed`)
```
Comment (robertkiel) [inline tests/node.ts:182]: Would rather say `dialling XYZ failed`

Comment (andreykiryushkin) [reply tests/node.ts:182]: can be confused with `dial` action above

**Review by robertkiel — APPROVED** [review]
lgtm, but need small merge with changes from relay management API

**Pull request merged**